external/kafka/src/main/scala/org/apache/spark/streaming/kafka/KafkaUtils.scala

```diff
@@ -154,6 +154,19 @@ object KafkaUtils {
       jssc.ssc, kafkaParams.toMap, Map(topics.mapValues(_.intValue()).toSeq: _*), storageLevel)
   }
 
+  /** get leaders for the given offset ranges, or throw an exception */
+  private def leadersForRanges(
+      kafkaParams: Map[String, String],
+      offsetRanges: Array[OffsetRange]): Map[TopicAndPartition, (String, Int)] = {
+    val kc = new KafkaCluster(kafkaParams)
```

Comment:
We keep creating a KafkaCluster object everywhere... what is the cost of creating a KafkaCluster object? Is there any? I guess not because it is just creating a config object and nothing else... isnt it?